### PR TITLE
[updated] Trigger at next available schedule

### PR DESCRIPTION
**Pull request by danielnorberg**
# Hey, I just made a Pull Request!

## Description
Continuation of https://github.com/spotify/styx/pull/565
Addresses #545

## Checklist for PR author(s)

- [x] Changes are covered by unit test
- [x] All tests pass
- [x] Code coverage check passes
- [ ] Error handling is tested
- [ ] Errors are handled at the appropriate layer
- [ ] Errors that cannot be handled where they occur are propagated
- [ ] (optional) Changes are covered by system test
- [ ] Relevant documentation updated
- [ ] This PR has NO breaking change to public API
- [ ] This PR has breaking change to public API and it is documented

## Checklist for PR reviewer(s)

- [ ] This PR has been incorporated in release note for the coming version
- [ ] Risky changes introduced by this PR have been all considered

### CODE REVIEW

**Comment by honnix** [comment]
Minor comments, otherwise LGTM.

**Comment by honnix** [comment]
Please also check whether https://github.com/spotify/styx/pull/651/commits/803115587da4b8a90cfcd7cf94bfe348cf378b07 makes sense.

File: styx-api-service/src/main/java/com/spotify/styx/api/workflow/WorkflowInitializer.java [hunk 1]
```diff
@@ -99,9 +99,10 @@ public WorkflowInitializer(Storage storage, Time time) {
 
   private TriggerInstantSpec initializeNaturalTrigger(Workflow workflow) {
     final Instant now = time.get();
+    final Instant offsetNow = workflow.configuration().subtractOffset(now);
     final Schedule schedule = workflow.configuration().schedule();
-    final Instant nextTrigger = lastInstant(now, schedule);
-    final Instant nextWithOffset = workflow.configuration().addOffset(nextTrigger);
+    final Instant nextTrigger = nextInstant(offsetNow, schedule); // Next trigger instant, rounded off by schedule
+    final Instant nextWithOffset = workflow.configuration().addOffset(nextTrigger); // gives offset by date
```
[the remark this answers is not in the document]
Comment (danielnorberg) [reply styx-api-service/src/main/java/com/spotify/styx/api/workflow/WorkflowInitializer.java]: Not sure these comments make sense to me.

File: styx-scheduler-service/src/test/java/com/spotify/styx/StyxSchedulerServiceFixture.java [hunk 2]
```diff
@@ -311,10 +311,12 @@ void workflowDeleted(Workflow workflow) throws IOException {
...
 
   private TriggerInstantSpec initializeNaturalTrigger(Workflow workflow) {
+    // TODO: duplicate of WorkflowInitializer.initializeNaturalTrigger
     final Instant now = time.get();
+    final Instant offsetNow = workflow.configuration().subtractOffset(now);
     final Schedule schedule = workflow.configuration().schedule();
-    final Instant nextTrigger = lastInstant(now, schedule);
-    final Instant nextWithOffset = workflow.configuration().addOffset(nextTrigger);
+    final Instant nextTrigger = nextInstant(offsetNow, schedule); // Next trigger instant, rounded off by schedule
+    final Instant nextWithOffset = workflow.configuration().addOffset(nextTrigger); // gives offset by date
```
[the remark this answers is not in the document]
Comment (danielnorberg) [reply styx-scheduler-service/src/test/java/com/spotify/styx/StyxSchedulerServiceFixture.java]: Removed

**Pull request merged**